[BUG]: Changes to pg driver in 0.45.0 cause startup error on Cloudflare Workers

### Report hasn't been filed before.

- [x] I have verified that the bug I'm about to report hasn't been filed before.

### What version of `drizzle-orm` are you using?

0.45.0

### What version of `drizzle-kit` are you using?

/

### Other packages

_No response_

### Describe the Bug

Works fine on `drizzle-orm@0.44.7`. 

This was probably added in https://github.com/drizzle-team/drizzle-orm/pull/1708. 

Simply importing `drizzle` should be enough to trigger the bug:

```
import { drizzle } from "drizzle-orm/node-postgres";
```

Startup error (`wrangler dev`):
``` 
⎔ Starting local server...
✘ [ERROR] service core:user:still-paper-b225: Uncaught Error: Dynamic require of "pg-native" is not supported

    at null.<anonymous> (index.js:12:9)
    at null.<anonymous> (index.js:6197:16) in
  node_modules/.pnpm/pg@8.16.3/node_modules/pg/lib/native/client.js
    at null.<anonymous> (index.js:18:50) in __require2
    at null.<anonymous> (index.js:6438:22) in
  node_modules/.pnpm/pg@8.16.3/node_modules/pg/lib/native/index.js
    at null.<anonymous> (index.js:18:50) in __require2
    at null.<anonymous> (index.js:6493:29) in get
    at null.<anonymous> (index.js:12719:30)


✘ [ERROR] The Workers runtime failed to start. There is likely additional logging output above.
```

Repro: https://github.com/tobimori/drizzle-45-cf-repro

--- a/changelogs/drizzle-orm/0.45.1.md
+++ b/changelogs/drizzle-orm/0.45.1.md
@@ -1,0 +1,1 @@
+- Fixed pg-native Pool detection in node-postgres transactions breaking in environments with forbidden `require()` ([#5107](https://github.com/drizzle-team/drizzle-orm/issues/5107))  

--- a/drizzle-orm/package.json
+++ b/drizzle-orm/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "drizzle-orm",
-	"version": "0.45.0",
+	"version": "0.45.1",
 	"description": "Drizzle ORM package for SQL databases",
 	"type": "module",
 	"scripts": {

--- a/drizzle-orm/src/node-postgres/session.ts
+++ b/drizzle-orm/src/node-postgres/session.ts
@@ -15,7 +15,6 @@ import { tracer } from '~/tracing.ts';
 import { type Assume, mapResultRow } from '~/utils.ts';
 
 const { Pool, types } = pg;
-const NativePool = (<any> pg).native ? (<{ Pool: typeof Pool }> (<any> pg).native).Pool : undefined;
 
 export type NodePgClient = pg.Pool | PoolClient | Client;
 
@@ -250,8 +249,9 @@ export class NodePgSession<
 		transaction: (tx: NodePgTransaction<TFullSchema, TSchema>) => Promise<T>,
 		config?: PgTransactionConfig | undefined,
 	): Promise<T> {
-		const session = (this.client instanceof Pool || (NativePool && this.client instanceof NativePool)) // eslint-disable-line no-instanceof/no-instanceof
-			? new NodePgSession(await this.client.connect(), this.dialect, this.schema, this.options)
+		const isPool = this.client instanceof Pool || Object.getPrototypeOf(this.client).constructor.name.includes('Pool'); // eslint-disable-line no-instanceof/no-instanceof
+		const session = isPool
+			? new NodePgSession(await (<pg.Pool> this.client).connect(), this.dialect, this.schema, this.options)
 			: this;
 		const tx = new NodePgTransaction<TFullSchema, TSchema>(this.dialect, session, this.schema);
 		await tx.execute(sql`begin${config ? sql` ${tx.getTransactionConfigSQL(config)}` : undefined}`);
@@ -263,9 +263,7 @@ export class NodePgSession<
 			await tx.execute(sql`rollback`);
 			throw error;
 		} finally {
-			if (this.client instanceof Pool || (NativePool && this.client instanceof NativePool)) { // eslint-disable-line no-instanceof/no-instanceof
-				(session.client as PoolClient).release();
-			}
+			if (isPool) (session.client as PoolClient).release();
 		}
 	}
 

